util/regexp-assemble/data/942130.data

```diff
@@ -23,19 +23,54 @@
 
 ##!+ i
 
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?=[\s'\"`()]*?(?:\b\1\b)
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?<=>[\s'\"`()]*?(?:\b\1\b)
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?like[\s'\"`()]*?(?:\b\1\b)
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?rlike[\s'\"`()]*?(?:\b\1\b)
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?sounds\s+like[\s'\"`()]*?(?:\b\1\b)
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?regexp[\s'\"`()]*?(?:\b\1\b)
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?!=[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?<=[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?>=[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?<>[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?<[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?>[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?\^[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?is\s+not[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?not\s+like[\s'\"`()]*?(?!\b\1\b)[\d\w]+
-[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?not\s+regexp[\s'\"`()]*?(?!\b\1\b)[\d\w]+
+##! General comments:
+##!
+##! The idea behind this expressions is to capture simple logic based (un)equalities that
+##! are used to quickly test SQL Logic that always returns TRUE (hence the term "SQL Tautology").
+
+##! Prefix: captures the initial part that will be matched on the right hand side of the logicl construct.
+##! This goes into the first rule of the chain:
+##! [\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?
+
+##! These expressions try to match the logic using the operator,
+##! so when the operator targets a TRUE operation, the initial match
+##! should be present after the operator, logically meaning TRUE
+##!
+##! Examples:
+##! '1' = '1'
+##! 'f' like 'f'
+
+=[\s'\"`()]*?(?:\b%{tx.lhs_942130}\b)
+<=>[\s'\"`()]*?(?:\b%{tx.lhs_942130}\b)
+
+##! Like queries allow you to use wilcards: '%'
+
+like[\s'\"`()]*?(?:\b%?%{tx.lhs_942130}%?\b)
+sounds\s+like[\s'\"`()]*?(?:\b%?%{tx.lhs_942130}%?\b)
+
+##! String based regexp. These don't use % as wildcard.
+rlike[\s'\"`()]*?(?:\b%{tx.lhs_942130}\b)
+regexp[\s'\"`()]*?(?:\b%{tx.lhs_942130}\b)
+
+##! These expressions try to match the logic using the negative operator,
+##! so when the operator targets a false operation, the initial match
+##! should *not* be present after the operator, effectively meaning TRUE
+##!
+##! Examples:
+##! '1' <= '2'
+##! 'a' not like 'b'
+
```

Comment:
All of these probably need to go into a separate rule where we use an inverted match (`!@rx`) instead of negative lookahead (`?!`). For re2 compatibility we need to get rid of lookarounds.